ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -776,6 +776,22 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			return err
 		}
 
+		// Delete all workers if worker group is suspended
+		if len(workerPods.Items) > 0 && worker.Suspend != nil && *worker.Suspend {
+			for _, workerPod := range workerPods.Items {
```

Comment:
Use `DeleteAllOf` instead?

Reply:
DeleteAllOf seems better, I didn't use it because of `r.rayClusterScaleExpectation.ExpectScalePod`. Do you know if it's okay to skip this call in this case?

Reply:
Updated to use `r.deleteAllPods` instead, which calls `DeleteAllOf`